File: day2/problem1.py
def is_safe(v):
    # Check if the list is sorted in ascending or descending order
    f = sorted(v)
    r = sorted(v, reverse=True)

    if v != f and v != r:
        return False
    
    # Check the differences between adjacent elements
    for i in range(1, len(v)):
        if not (0 < abs(v[i] - v[i - 1]) <= 3):
            return False
    
    return True

def main():
    ans = 0
    with open("input1.txt") as file:
        for line in file:
            v = list(map(int, line.split()))
            if is_safe(v):
                ans += 1
    
    print(ans)

if __name__ == "__main__":
    main()

File: day2/problem1.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

bool is_safe(vector<int> &v) {
    vector<int> f = v, r = v;
    sort(f.begin(), f.end());
    sort(r.begin(), r.end(), greater<int>());

    if (v != f && v != r) return false;

    for (int i = 1; i < v.size(); ++i) {
        if (abs(v[i] - v[i - 1]) <= 3 && abs(v[i] - v[i - 1]) > 0) continue;
        return false;
    }
    return true;
}

int main() {
    ifstream file("input1.txt");
    string line;
    ll ans = 0;

    while (getline(file, line)) {
        istringstream iss(line);
        vector<int> v;
        int x;

        while (iss >> x) v.push_back(x);

        if (is_safe(v)) {
            ans++;
        }
    }
    
    cout << ans << '\n';
    return 0;
}
